### PR TITLE
1412: Copy, Duplicate & Select Features

### PR DESCRIPTION
**Pull request by lilitkarapetyan**
## 🧰 Issue

#1412

## 🔨Work carried out:

- [x] Tests pass

## Confirmations

- [ ] I have chosen reviewers for my PR.
- [ ] I have assigned myself to this PR.
- [ ] I have chosen an appropriate label for the PR.
- [ ] I have completed the mandatory sections of this document.
- [ ] I have deleted any unused sections.
- [ ] I confirm that I have checked for required README updates and acted accordingly.

### CODE REVIEW

File: packages/client/src/api/wargames_api/index.ts [hunk 1]
```diff
@@ -386,6 +386,29 @@ export const deletePlatformType = (dbName: string, platformType: PlatformType):
   })
 }
 
+export const duplicatePlatformType = (dbName: string, currentPlatformType: PlatformType): Promise<Wargame> => {
+  return getLatestWargameRevision(dbName).then((res) => {
+    const newDoc: Wargame = deepCopy(res)
+    const updatedData = newDoc.data
+    const platformTypes = updatedData.platformTypes && updatedData.platformTypes.platformTypes || []
```
Comment (IanMayo) [inline packages/client/src/api/wargames_api/index.ts]: @lilitkarapetyan - on the next two lines we find the index of the platform, then duplicate the array item at that index.

Is there a reason why we're not using the single-line `platformTypes.find`? 

Aah, do we get the index so we can push the duplicate entry in on the next row in the list?

File: packages/client/src/api/wargames_api/index.ts [hunk 2]
```diff
@@ -386,6 +386,29 @@ export const deletePlatformType = (dbName: string, platformType: PlatformType):
...
+    const platformTypeIndex = platformTypes.findIndex((platformType) => platformType.name === currentPlatformType.name)
+    const duplicatedPlatformType = deepCopy(platformTypes[platformTypeIndex])
+    const uniq = uniqid.time()
+
+    duplicatedPlatformType.name = `${duplicatedPlatformType.name}-${uniq}`
+
+    platformTypes.splice(platformTypeIndex, 0, duplicatedPlatformType)
+
+    if (updatedData.platformTypes) {
```
Comment (IanMayo) [inline packages/client/src/api/wargames_api/index.ts]: Hmm, the whole block of code could go inside this `if`.  Then you won't need `updatedData.platformTypes &&` at line 393

File: packages/client/src/api/wargames_api/index.ts (hunk 1 above)
Comment (lilitkarapetyan) [reply packages/client/src/api/wargames_api/index.ts]: I did this to not change the duplication logic and it works like duplicating channels.

**Comment by IanMayo** [comment]
@lilitkarapetyan - could you have a look over my changes please/

**Comment by lilitkarapetyan** [comment]
> @lilitkarapetyan - could you have a look over my changes please/

Works great

**Review by IanMayo — APPROVED** [review]

**Comment by ShendMikullovci** [comment]
Just reviewed this. Works well on the Platform page. I can't save the forces for some reason so I can't try the feature on the Forces page.

**Pull request merged**